internal/clientconn/conninfo/conn_info.go

```diff
@@ -34,6 +36,16 @@ type ConnInfo struct {
 	rw       sync.RWMutex
 	username string
 	password string
+
+	curRW  sync.RWMutex
+	cursor map[string]iterator.Interface[int, any]
```

Comment:
It is possible to have multiple cursor in one connection for the same collection:

```
test> for (var i = 0; i < 1000; i++) { db.test.insertOne({}) }
…

test> c1 = db.test.find({})
[
  { _id: ObjectId("63c91563be6127738499f256") },
  …
]

test> c2 = db.test.find({})
[
  { _id: ObjectId("63c91563be6127738499f256") },
  …
]

test> c1.next()
{ _id: ObjectId("63c91563be6127738499f26a") }

test> c1.next()
{ _id: ObjectId("63c91563be6127738499f26b") }

test> c2.next()
{ _id: ObjectId("63c91563be6127738499f26a") }

test> c2.next()
{ _id: ObjectId("63c91563be6127738499f26b") }
```

Reply:
Let's use listener-scoped or handler-scoped atomic int64 as cursor ID and use it, not collection name

Reply:
# 1733 says:
> and closes it when another command is issued

So let's do just that and keep one cursor per connection

Reply:
Done.